pkg/apis/core/types.go

```diff
@@ -3316,6 +3316,14 @@ type ServiceSpec struct {
 	// of peer discovery.
 	// +optional
 	PublishNotReadyAddresses bool
+
+	// topologyKeys is a preference-order list of topology keys which clients should respect when accessing this Service.
+	// If any ready backends exist for index [0], they should always be chosen; only if no backends exist for index [0] should backends for index [1] be considered.
+	// If this field is specified, but all entries have no backends that match the topology of the client, the service has no backends for that client and connections should fail.
+	// The special value "" may be used to means "any node". This catch-all value, if used, only makes sense as the last value in the list.
+	// If this is not specified or empty, no topoly contraints will be applied.
+	// +optional
+	TopologyKeys []string
```

Comment:
You need to add validation (flag-gated) to verify that these are valid label keys or the special value.

Reply:
Yes, I'll add the validation for `TopologyKeys` in `ValidateService` (flag-gated) :
https://github.com/kubernetes/kubernetes/blob/117e83157bc340c1823f5bd34d4f73ca126ccdd6/pkg/apis/core/validation/validation.go#L3900

Reply:
done